File: problems/409_longest_palindrome.py
class Solution:
    def longestPalindrome(self, s: str) -> int:
        counts: dict[str, int] = {}
        for char in s:
            counts[char] = counts.get(char, 0) + 1
        ans = 0
        for count in counts.values():
            if count >= 2:
                ans += count - count % 2
        if len(s) > ans:
            ans += 1
        return ans

File: problems/409-longest-palindrome.ts
function longestPalindrome(s: string): number {
  const charCounts = new Map<string, number>()
  for (const char of s) {
    charCounts.set(char, (charCounts.get(char) ?? 0) + 1)
  }
  let ans = 0
  for (const count of charCounts.values()) {
    if (count >= 2) {
      ans += count - (count % 2)
    }
  }
  return s.length > ans ? ans + 1 : ans
}
